mountpoint-s3/src/main.rs

```diff
@@ -430,13 +441,30 @@ fn mount(args: CliArgs) -> anyhow::Result<FuseSession> {
 
     validate_mount_point(&args.mount_point)?;
 
-    let addressing_style = args.addressing_style();
+    let supposed_region = &args.region;
+    const DEFAULT_REGION: &str = "us-east-1";
+
+    let region_to_try = supposed_region.clone().unwrap_or_else(|| {
+        tracing::warn!("region unspecified. using {} as the signing region.", DEFAULT_REGION);
```

Comment:
I can add the condition of endpoint URI provided here. But, I have added this region configuration here instead of `create_client_for_bucket` because there is no default choice for region in `EndpointConfig`. So, we either have to provide a default region here or build the `EndpointConfig` within `create_client_for_bucket` which will require passing CliArgs to `create_client_for_bucket`.

Reply:
I think the conversion from CliArgs to EndpointConfig is complicated enough that we should factor it out into its own function.